src/qt/bitcoin.cpp

```diff
@@ -282,6 +283,7 @@ void BitcoinCore::initialize()
     try
     {
         qDebug() << __func__ << ": Running initialization in thread";
+        thread_util::Rename("bitcoin-qt");
```

Comment:
In commit "threads: explicitly name main threads" (56deb18c8bc36de5bb49c7b1db27adfa80ce7169)

This is giving 2 different threads (the `main()` thread, and the initialization thread spawned in `BitcoinApplication::startThread`) the same name. It might be better to distinguish this thread by calling it `qt-init` or something instead of `bitcoin-qt`.

This is also kind of an awkward place to be name the thread, since it isn't in a standard thread entry point or close to where the thread is created. Maybe it would make sense for us to have a custom QThread subclass that supports named threads, and calls `thread_util::Rename` in [`exec()`](http://doc.qt.io/qt-5/qthread.html#exec). Or maybe we could have write a little helper object that attaches to a `QThread`, listens for the [`started`](http://doc.qt.io/qt-5/qthread.html#started) signal, and sets the thread name in the signal handler.

Reply:
Changed the name of this thread to `bitcoin-qt-init`.

Agree with you that this is an awkward spot for naming. Do you think your proposed fix should be done along with this changeset or maybe as a follow-up PR?

Reply:
https://github.com/bitcoin/bitcoin/pull/13168#discussion_r195165776

> Do you think your proposed fix should be done along with this changeset or maybe as a follow-up PR?

I might opt to do it in a different PR because QThread is also used in rpcconsole.cpp and bitcoin.cpp, and it might make sense to update those places to set thread names in the same way. But I don't have a real opinion. It seems fine to make the change in this PR, in another PR, or not at all.